allow for host build of tools enabling cross compilation

heitbaum opened the pull request.

heitbaum committed:
allow for host build of tools enabling cross compilation

antoniovazquezblanco commented:
Thank you @heitbaum!

I am looking at your patch and I would like to request a couple of changes. Up to you if you want to split the following suggestions into separate PRs or to do it all at once :).

1. Please, lets keep the options file well organized. First, tool options and then source options or the other way around. Separate both sections with a comment. That makes it extra nice!

2. Regarding option naming general conventions, I do not have any hard rule here. Now seing the options file, it seems that `enable-string-pooling` may not be the most conventional thing knowing that meson mostly uses underscores (https://mesonbuild.com/Build-options.html). This may be inherited from the config file (I do not really remember). Maybe, in newer options we should stick to meson standards and use underscores. I like very explicit names such as `enable_something` rather than `something` or `tools_search_path` rather than `tools`.

3. Regarding the tool options:
    a. There is already an `enable-string-pooling` option that is related to the options you have added. It is mostly handled inside the `tools/strpool/meson.build`. I think I would be happier if the new tooling related options are handled there... That would make more clear situations as I will explain in b.
    b. If `enable-string-pooling` is set to false, then there should be no need to look for the strpool utility, and right now, it seems that may happen and fail if not found...
    c. Can we use `enable_strpool_build` instead of the generic `build-tools`? If not build and strpooling is enabled, then we search for it.
    d. I would also be happier if the `strpool` option name was a little bit more explicit. I am unsure of what the best practice would be if either accepting search paths or accepting tool path directly. Up to you :)
    
 Sorry if I am being too verbose. Also, sometimes my way of reviewing things may be a little bit chaotic.
 
 Thanks again for the PR! :)

heitbaum commented:
> 1. Please, lets keep the options file well organized. First, tool options and then source options or the other way around. Separate both sections with a comment. That makes it extra nice!

Ordered sensibly (and in sort order)

> 2. Regarding option naming general conventions, I do not have any hard rule here. Now seing the options file, it seems that `enable-string-pooling` may not be the most conventional thing knowing that meson mostly uses underscores (https://mesonbuild.com/Build-options.html). This may be inherited from the config file (I do not really remember). Maybe, in newer options we should stick to meson standards and use underscores. I like very explicit names such as `enable_something` rather than `something` or `tools_search_path` rather than `tools`.

Fixed. I have updated all to underscore. I have updated to house style `enable_` but personally wouldn’t use that style, that is more an olds configure style, whereas meson examples and code I have maintained doesn’t have this; as everything is declarative =true or =false.

> 3. Regarding the tool options:

Reworked.

antoniovazquezblanco commented:
> I have updated to house style enable_ but personally wouldn’t use that style

That was also inherited. I will merge as is but will remove the enable prefix as per your suggestion.

Thanks @heitbaum for your contribution!
